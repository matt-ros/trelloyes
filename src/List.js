import React from 'react';
import Card from './Card';
import './List.css';

function List(props) {
    const cardList = props.cards.map((card) => 
    <Card key={card.id} id={card.id} title={card.title} content={card.content} onDeleteCard={props.onDeleteCard} />
    );

    return (
        <section className="List">
            <header className="List-header">
                <h2>{props.header}</h2>
            </header>
            <div className="List-cards">
                {cardList}
                <button type="button" className="List-add-button" onClick={() => props.onRandomCard(props.id)}>
                    + Add Random Card
                </button>
            </div>
        </section>
    )
}

export default List;