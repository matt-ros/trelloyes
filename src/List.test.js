import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Card from './Card';
import List from './List';

describe('List component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<List header="test header" cards={[
            {id: "test-key", title: "test title", content: "test content"}
            ]}
        />,
        div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<List header="test header" cards={[
                {id: "test-key-1", title: "test title 1", content: "test content 1"},
                {id: "test-key-2", title: "test title 2", content: "test content 2"}
            ]}
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});