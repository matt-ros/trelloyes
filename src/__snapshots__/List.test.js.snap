// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      test header
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        test title 1
      </h3>
      <p>
        test content 1
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        test title 2
      </h3>
      <p>
        test content 2
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
